File: other/tutorial/segmentation3.py
import sys
import os
PathPop= ""
if os.path.isfile(PathPop+"population.py")==0:
    print "set the variable PathPop to the path where you compile population, for instance D:\Users/vtariel/Desktop/ANV/Population-build/. This folder must contain population.py"
    sys.exit(-1)
sys.path.append(PathPop)
from population import * 
    

img = Mat2UI8()
img.load(PathPop+"/image/iex.png")
proc = Processing()
filter = proc.smoothDeriche(img,1)
filter = proc.dynamic(filter,40)
minima = proc.minimaRegional(filter)
visu = Visualization()
#	visu.labelForeground(minima,img).display()
water = proc.watershedBoundary(minima,filter,1)
boundary = proc.threshold(water,0,0)#the boundary label is 0
#boundary.display("boundary",true,false)
minima = proc.labelMerge(Mat2UI32(boundary),minima)#the pixel type of the image must be the same f.
gradient = proc.gradientMagnitudeDeriche(img,1)
water = proc.watershed(minima,gradient)
visu.labelForeground(water,img).display()

File: other/tutorial/segmentation3.cpp
#include"Population.h"//Single header
using namespace pop;//Population namespace
int main(){

    Mat2UI8 img;
    img.load(POP_PROJECT_SOURCE_DIR+std::string("/image/iex.png"));
    Mat2UI8 filter = Processing::smoothDeriche(img,1);
    filter = Processing::dynamic(filter,40);
    Mat2UI16 minima = Processing::minimaRegional(filter);
    //        Visualization::labelForeground(minima,img).display();
    Mat2UI16 water =Processing::watershedBoundary(minima,filter,1);
    Mat2UI16 boundary = Processing::threshold(water,0,0);//the boundary label is 0
    //        boundary.display("boundary",true,false);
    minima = Processing::labelMerge(boundary,minima);
    Mat2UI8 gradient = Processing::gradientMagnitudeDeriche(img,1);
    water = Processing::watershed(minima,gradient);
    Visualization::labelForeground(water,img).display();

}
